api/v1beta1/nutanixfailuredomain_types.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2025 Nutanix
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+const (
+	// NutanixFailureDomainKind represents the Kind of NutanixFailureDomain
+	NutanixFailureDomainKind = "NutanixFailureDomain"
+
+	// NutanixFailureDomainFinalizer is the finalizer used by the NutanixFailureDomain controller to block
+	// deletion of the NutanixFailureDomain object if there are references to this object by other resources.
+	NutanixFailureDomainFinalizer = "nutanixfailuredomain.infrastructure.cluster.x-k8s.io"
```

Comment:
nit: we should consider using a domain qualified finalizer e.g `infrastructure.cluster.x-k8s.io/nutanixfailuredomain` instead
see: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#finalizers